ddl/index.go

```diff
@@ -182,15 +186,13 @@ func getIndexColumnLength(col *model.ColumnInfo, colLen int) (int, error) {
 	switch col.Tp {
 	case mysql.TypeBit:
 		return (length + 7) >> 3, nil
-	case mysql.TypeVarchar, mysql.TypeString:
+	case mysql.TypeVarchar, mysql.TypeString, mysql.TypeTinyBlob, mysql.TypeMediumBlob, mysql.TypeBlob, mysql.TypeLongBlob:
```

Comment:
I think the blob type has nothing to do with charset?

Reply:
right, but we tread `text` as `blob` with charset, the blob can only be `binary` charset and its `maxlen` is 1, so that here is nothing different from the original.

Reply:
But the old code has the better performance.